### PR TITLE
Attribution

### PR DESCRIPTION
**Pull request by jeegarshah**
New components Field, TextField, ChoiceInput, DialogLayout with docs

### CODE REVIEW

File: packages/visual-stack/src/components/Form.js [hunk 1]
```diff
@@ -44,3 +44,31 @@ FormGroup.propTypes = {
   label: PropTypes.node,
   required: PropTypes.bool,
 };
+
+// Component that renders a label and error/help message
+export const Field = ({label, error, help, optional, optionalLabel = "Optional", className, children}) => (
```
Comment (fhwrdh) [inline packages/visual-stack/src/components/Form.js]: Do we have a consistent approach to default label text? I see "Optional" here, I see other labels (DialogLayout buttons, for instance) without defaults.

File: packages/visual-stack-docs/src/containers/App/SideNav.js [hunk 2]
```diff
@@ -22,13 +22,20 @@ import {
...
 import { routeComponentMap } from '../Components/Docs/';
+import { layoutsRouteMap } from '../Layouts';
 
 const componentLinks = R.pipe(
   R.mapObjIndexed((val, key) => ({ key, ...val })),
   R.values,
   R.sortBy(R.prop('linkName'))
 )(routeComponentMap);
 
+const layoutLinks = R.pipe(
```
Comment (fhwrdh) [inline packages/visual-stack-docs/src/containers/App/SideNav.js]: This looks to duplicate the componentLinks bit above. Can we extract the function here and reuse with the code above?

File: packages/visual-stack-docs/src/containers/Layouts/DialogLayout/index.js [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+
+                </div>
+            )}
+        </Demo>
+    );
+  }
+}
+
+/* s1:start */
+export class DialogLayoutDemo extends React.Component {
```
Comment (fhwrdh) [inline packages/visual-stack-docs/src/containers/Layouts/DialogLayout/index.js]: Any reason this isn't just a functional component?

File: packages/visual-stack-docs/src/containers/Layouts/DialogLayout/index.js [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+                    </Panel>
+                </DialogLayout>
+            </div>
+        );
+    }
+}
+/* s1:end */
+
+const NotRenderedComponent = () => (
+    <div>
```
Comment (fhwrdh) [inline packages/visual-stack-docs/src/containers/Layouts/DialogLayout/index.js]: Need a `<div>` wrapper here?

File: packages/visual-stack/src/components/Form.js (hunk 1 above)
Comment (jeegarshah) [reply packages/visual-stack/src/components/Form.js]: this is differentiating required vs optional field. The ui guideline says to show "- Optional" along the label. Now we will need translation anyway so we can take out the default. ? @fhwrdh

Comment (fhwrdh) [reply packages/visual-stack/src/components/Form.js]: Yes, we should probably take the default out here. For translation purposes, it should also eventually be a separate prop.

**Pull request merged**